### ISSUE TITLE
During build: string concatenation in url_for() fails when data comes from render_template()

### ISSUE THREAD

**Issue by LokiWijnen** [Frozen-Flask/Frozen-Flask#121]
When running Flask normally, everything described below works absolutely fine.

When running a build in Frozen-Flask however, I run into the issue where my dynamically built route string won't allow me to concatenate a variable (containing a string) with a string for use in a `url_for()` function within Jinja2. But only when the variable data is originating from Python such as `render_template()`.

routes.py
```python3
@app.route('/some-route/')
def some_route():
    return render_template('my_template.html', foo='bar')
```

base.html
```jinja2
{% import "includes.html" as includes %}
{{ includes.my_macro(foo) }}
```

includes.html
```jinja2
# The following does NOT work
{% macro my_macro(foo) %}
    <a href="{{ url_for('index_' ~ foo) }}">Link text</a>
{% endmacro %}
```

Although the above runs fine in Flask, when I do a build with Frozen-Flask I get the following error:

```cli
werkzeug.routing.BuildError: Could not build url for endpoint 'index_'. Did you mean...
```

As you can see, the value of foo (or the string value `bar`) is missing. It should have been `index_bar`.

So I tried this instead:

```jinja2
# This also does NOT work
{% macro my_macro(foo) %}
    {% set route = 'index_' ~ foo %}
    <a href="{{ url_for(route) }}">Link text</a>
{% endmacro %}
```

The above produces the exact same error.

So I tried this to try to better understand the problem:

```jinja2
# This works correctly
{% macro my_macro(foo) %}
    {% set route = 'index_' ~ 'bar' %}
    <a href="{{ url_for(route) }}">Link text</a>
{% endmacro %}
```

So I further tried this:

```jinja2
# This also works correctly
{% macro my_macro(foo) %}
    {% set foo2 = 'bar' %}
    {% set route = 'index_' ~ foo2 %}
    <a href="{{ url_for(route) }}">Link text</a>
{% endmacro %}
```

So basically, I can create a dynamic route string for use by `url_for()` but only when the variable data doesn't come from the Flask routes file (such as my `routes.py`), only if the variable is created within Jinja2.

**Comment by TomHall2020** [comment]
Your routes.py file in the example only defines `some_route`. I'm trying to understand your setup here, do you actually have multiple routes with endpoints `index_bar`, `index_other` etc? If so you may want to create a true dynamic route as follows:

```
@app.route('/somewhere/<foo>')
def somewhere(foo):
    return render_template('my_template.html', foo=foo)
```

then in your templates you can access these via url_for as `url_for(somewhere, foo='bar')` and skip out the macro and string concatenation in the template.

**Comment by liZe** [comment]
I can’t reproduce this bug anymore. If someone has a full app example that doesn’t work, please add a comment!
